core/templates/dev/head/components/CreateActivityButtonDirective.js

```diff
@@ -44,29 +44,34 @@ oppia.directive('createActivityButton', [function() {
           if (urlService.getPathname() !== '/my_explorations') {
             $window.location.replace('/my_explorations?mode=create');
           } else {
-            $modal.open({
-              templateUrl: 'modals/createActivity',
-              backdrop: true,
-              controller: [
-                  '$scope', '$modalInstance', function($scope, $modalInstance) {
-                $scope.chooseExploration = function() {
-                  ExplorationCreationService.createNewExploration();
-                  $modalInstance.close();
-                };
+            if ($('.oppia-create-activity-modal').length === 0) {
```

Comment:
More specifically, what I did was to move `$scope.creationInProgress` into after the if statement (that you highlighted, instead of outside) and had that check on the variable.